src/main/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTask.kt

```diff
@@ -440,44 +442,69 @@ open class RunPluginVerifierTask @Inject constructor(
      * @return path to the Java Runtime directory
      */
     private fun resolveRuntimeDir(): String {
-        runtimeDir.orNull?.let {
-            debug(context, "Runtime specified with properties: $it")
-            return it
-        }
-
         val jbrResolver = objectFactory.newInstance(
             JbrResolver::class.java,
             extension.jreRepository.orNull ?: "",
             isOffline,
             this,
         )
-        jbrVersion.orNull?.let {
-            jbrResolver.resolve(jbrVersion.orNull)?.let { jbr ->
-                debug(context, "Runtime specified with JBR Version property: $it")
-                return jbr.javaHome.canonicalPath
-            }
-            warn(context, "Cannot resolve JBR $it. Falling back to built-in JBR.")
-        }
 
         val jbrPath = when (OperatingSystem.current().isMacOsX) {
             true -> "jbr/Contents/Home"
             false -> "jbr"
         }
 
-        getBuiltinJbrVersion(ideDir.get().asFile)?.let { builtinJbrVersion ->
-            jbrResolver.resolve(builtinJbrVersion)?.let { builtinJbr ->
-                val javaHome = File(builtinJbr.javaHome, jbrPath)
-                if (javaHome.exists()) {
-                    debug(context, "Using built-in JBR: $javaHome")
-                    return javaHome.canonicalPath
+        return listOf(
+            {
+                runtimeDir.orNull
+                    ?.also { debug(context, "Runtime specified with properties: $it") }
+            },
+            {
+                jbrResolver.resolve(jbrVersion.orNull)?.javaHome?.canonicalPath
+                    ?.also { debug(context, "Runtime specified with JBR Version property: ${jbrVersion.get()}") }
```

Comment:
.get() or not get() (next line)

Reply:
First – can be `null`, that's why I call `.getOrNull()`, second – we already know it's not null.